### PR TITLE
Changes to Gosnappi Data type refactor changes

### PR DESCRIPTION
**Pull request by Azharkuntoji**
Modified all the tests in the main for data type refactor changes.
Gosnappi version github.com/open-traffic-generator/snappi/gosnappi@v0.12.1 has data type refactor changes.

### CODE REVIEW

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]
@Azharkuntoji : Please merge with the latest from the main branch and also update go.mod with the latest gosnappi version (and subsequently update go.sum as well.)

**Review by jasdeep-hundal — APPROVED** [review]

**Comment by Azharkuntoji** [comment]
@jasdeep-hundal resolved a test change from the main. Please review it again.

File: topologies/kne/arista/ceos/dutate.textproto [hunk 1]
```diff
@@ -35,45 +35,3 @@ links: {
...
-    a_int: "eth8"
-    z_node: "otg"
-    z_int: "eth8"
-}
-links: {
-    a_node: "dut"
-    a_int: "eth9"
-    z_node: "otg"
-    z_int: "eth9"
-}
```
Comment (jasdeep-hundal) [inline topologies/kne/arista/ceos/dutate.textproto:79]: Why are these lines being deleted?

Comment (Azharkuntoji) [reply topologies/kne/arista/ceos/dutate.textproto]: deleted by mistake. I have reverted the changes

**Review by jasdeep-hundal — APPROVED** [review]

**Pull request merged**